spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/actions/TestExpireSnapshotsAction.java

```diff
@@ -1176,10 +1176,10 @@ public void testExpireAction() {
 
     Assert.assertEquals("Should not delete any files", 0, deletedFiles.size());
 
-    Assert.assertSame(
-        "Multiple calls to expire should return the same deleted files",
-        pendingDeletes,
-        action.expire());
+    Assert.assertEquals(
```

Comment:
This no longer has to be the same reference. I reuse `QueryExecution` so we keep the old behavior. I added a new test below. It would fail otherwise.